src/main/java/core/basesyntax/service/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        OperationHandler handler = operationHandlers.get(transaction.getOperation());
+        if (handler != null) {
+            handler.apply(transaction);
+        } else {
+            throw new RuntimeException("Operation handler not found for operation: "
+                    + transaction.getOperation());
+        }
```

Comment:
in cases like this, invert if condition, throw early - it's a guard pattern, preferrable approach, because you won't need else after returning/throwing
https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement